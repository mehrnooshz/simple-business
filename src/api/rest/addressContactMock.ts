export interface Props {
  title: string;
  descriptionFirst: string;
  descriptionSecond: string;
}

const addressContactMock: Props = {
  title: 'test title',
  descriptionFirst: 'first description',
  descriptionSecond: 'second description',
};

export default addressContactMock;
