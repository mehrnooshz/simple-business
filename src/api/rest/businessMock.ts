import Business from 'api/model/business';

export const BusinessMock = {
  name: 'test',
  id: '1',
  description: 'hi',
  phone: '0911',
  image: 'url',
  email: 'a@yahoo.com',
  address: {
    number: '021',
    street: ' book',
    zip: '123',
    city: 'rasht',
    country: 'iran',
  },
};
export const BusinessSameMock = {
  name: 'test2',
  id: '2',
  description: 'hi',
  phone: '0911',
  image: 'url2',
  email: 'b@yahoo.com',
  address: {
    number: '011',
    street: 'cover',
    zip: '456',
    city: 'rasht',
    country: 'iran',
  },
};

export const BusinessDifferentMock = {
  name: 'test3',
  id: '3',
  description: 'hi',
  phone: '0911',
  image: 'url2',
  email: 'c@yahoo.com',
  address: {
    number: '011',
    street: 'cover',
    zip: '456',
    city: 'sofia',
    country: 'bulgharia',
  },
};

export const BusinessesMock: Business[] = [
  BusinessMock,
  BusinessSameMock,
  BusinessDifferentMock,
];

export const businessIdExistedMock: string = '1';
export const businessIdNotExistedMock: string = '4';
