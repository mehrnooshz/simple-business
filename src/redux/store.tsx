import {
  applyMiddleware,
  combineReducers,
  createStore,
  Store,
  AnyAction,
} from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import AppState from './state/appState';
import BusinessListReducer from './reducers/businessReducer';

const rootReducer = combineReducers<AppState>({
  businessListState: BusinessListReducer,

  // TODO: add sub-reducers here...
});

export default function configureStore(): Store<AppState, AnyAction> {
  return createStore(
    rootReducer,
    undefined,
    composeWithDevTools(applyMiddleware(thunk))
  );
}
