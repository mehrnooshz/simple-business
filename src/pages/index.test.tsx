import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import renderer, {
  ReactTestRenderer,
  ReactTestInstance,
} from 'react-test-renderer';
import { BusinessesMock } from 'api/rest/businessMock';
import App from './';
import AppState from 'redux/state/appState';
import { BrowserRouter } from 'react-router-dom';
import { Store } from 'redux';
import ItemList from 'components/business/itemList';
import { cleanup } from '@testing-library/react';

describe('App', () => {
  const createMockStore = configureStore<AppState>([thunk]);

  const createEmptyState = (): AppState => {
    return {
      businessListState: {
        businesses: [],
        sameBusinesses: [],
        business: undefined,
        isFetching: false,
      },
    };
  };

  const createTestRenderer = (store: Store<AppState>): ReactTestRenderer => {
    return renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </BrowserRouter>
    );
  };

  describe('should be rendered', () => {
    it('inside a redux Provider', () => {
      createTestRenderer(createMockStore(createEmptyState()));
    });

    it('and it should match the expected snapshot', () => {
      expect(
        createTestRenderer(createMockStore(createEmptyState())).toJSON()
      ).toMatchSnapshot();
    });

    describe('if business exist check props of itemList', () => {
      const businesses = BusinessesMock;
      const isFetching = false;
      const error = undefined;
      const state = createEmptyState();
      state.businessListState.businesses = businesses;
      state.businessListState.isFetching = isFetching;
      state.businessListState.error = error;

      let component: ReactTestRenderer;
      let itemList: ReactTestInstance;

      beforeEach(() => {
        component = createTestRenderer(createMockStore(state));
        itemList = component.root.findByType(ItemList);
      });
      it('check business props in itemList', () => {
        expect(itemList.props.businesses).toBe(businesses);
      });
      it('check isFetching props in itemList', () => {
        expect(itemList.props.isFetching).toBe(isFetching);
      });
      it('check error props in itemList', () => {
        expect(itemList.props.error).toBe(error);
      });
      afterEach(() => {
        cleanup();
      });
    });
  });
});
