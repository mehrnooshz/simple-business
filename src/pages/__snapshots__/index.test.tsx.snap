// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should be rendered and it should match the expected snapshot 1`] = `
<div
  className="container"
>
  <ul
    className="responsive-table bg-white"
  >
    <li
      className="table-header d-flex"
    >
      <div
        className="col-3 title"
      >
        NAME
      </div>
      <div
        className="col-7 title"
      >
        DESCRIPTION
      </div>
    </li>
  </ul>
</div>
`;
